crates/sequencer/src/commitment/controller.rs

```diff
@@ -0,0 +1,161 @@
+use std::cmp;
+
+use citrea_common::utils::merge_state_diffs;
+use citrea_primitives::compression::compress_blob;
+use citrea_primitives::MAX_TXBODY_SIZE;
+use sov_db::ledger_db::SequencerLedgerOps;
+use sov_db::schema::types::BatchNumber;
+use sov_modules_api::StateDiff;
+use tracing::{debug, warn};
+
+use super::CommitmentInfo;
+
+pub struct CommitmentController<Db>
+where
+    Db: SequencerLedgerOps,
+{
+    ledger_db: Db,
+    min_soft_confirmations: u64,
+    last_state_diff: StateDiff,
+}
+
+impl<Db> CommitmentController<Db>
+where
+    Db: SequencerLedgerOps,
+{
+    pub fn new(ledger_db: Db, min_soft_confirmations: u64) -> Self {
+        let last_state_diff = ledger_db.get_state_diff().unwrap_or_default();
+        Self {
+            ledger_db,
+            min_soft_confirmations,
+            last_state_diff,
+        }
+    }
+
+    pub fn should_commit(
+        &mut self,
+        l2_height: u64,
+        l2_state_diff: StateDiff,
```

Comment:
Why not &StateDiff?

Reply:
where we call should_commit, state diff is only needed once, so we can just give the ownership to this method